lib/cluster/subscriber.js

```diff
@@ -0,0 +1,680 @@
+      broadcast: false,
+      reason: 'Node is shutting down',
+    });
+  }
+
+  /**
+   * Handles messages about realtime room creations
+   *
+   * @param  {Object} message - decoded NewRealtimeRoom protobuf message
+   * @return {void}
+   */
+  handleNewRealtimeRoom (message) {
+    const { roomId, index, collection, filters, messageId } = message;
+
+    debug(
+      'New realtime room created by node %s (message: %d, room: %s, index: %s, collection: %s)',
+      this.remoteNodeId,
+      messageId,
+      roomId,
+      index,
+      collection);
+
+    this.localNode.fullState.addRealtimeRoom(
+      roomId,
+      index,
+      collection,
+      JSON.parse(filters),
+      {
+        messageId,
+        nodeId: this.remoteNodeId,
+        subscribers: 0,
+      });
+  }
+
+  /**
+   * Handles messages about realtime subscriptions
+   *
+   * @param  {Object} message - decoded Subscription protobuf message
+   * @return {void}
+   */
+  handleSubscription (message) {
+    debug('New realtime subscription received from node %s (message: %d, room: %s)',
+      this.remoteNodeId,
+      message.messageId,
+      message.roomId);
+
+    this.localNode.fullState.addRealtimeSubscription(
+      message.roomId,
+      this.remoteNodeId,
+      message.messageId);
+  }
+
+  /**
+   * Handles messages about realtime room removal
+   *
+   * @param  {Object} message - decoded RemoveRealtimeRoom protobuf message
+   * @return {void}
+   */
+  handleRealtimeRoomRemoval (message) {
+    debug('Realtime room removal received from node %s (message: %d, room: %s)',
+      this.remoteNodeId,
+      message.messageId,
+      message.roomId);
+
+    this.localNode.fullState.removeRealtimeRoom(message.roomId, this.remoteNodeId);
+  }
+
+  /**
+   * Handles messages about user unsubscriptions
+   *
+   * @param  {Object} message - decoded Unscription protobuf message
+   * @return {void}
+   */
+  handleUnsubscription (message) {
+    debug('Realtime unsubscription received from node %s (message: %d, room: %s)',
+      this.remoteNodeId,
+      message.messageId,
+      message.roomId);
+
+    this.localNode.fullState.removeRealtimeSubscription(
+      message.roomId,
+      this.remoteNodeId,
+      message.messageId);
+  }
+
+  /**
+   * Handles messages about cluster-wide events
+   *
+   * @param {Object} message - decoded ClusterWideEvent protobuf message
+   * @return {void}
+   */
+  async handleClusterWideEvent (message) {
+    const payload = JSON.parse(message.payload);
+
+    this.localNode.eventEmitter.emit(message.event, payload);
+  }
+
+  /**
+   * Handles messages about document notifications
+   *
+   * @param {Object} message - decoded DocumentNotification protobuf message
+   * @return {void}
+   */
+  async handleDocumentNotification (message) {
+    const notification = new DocumentNotification(
+      null,
+      message.scope,
+      message.action,
+      JSON.parse(message.result));
+
+    notification.node = this.remoteNodeId;
```

Comment:
You're right, I made a mistake in the current implementation, as it will trigger a de-opt from V8, since the instantiated class won't have the same properties whether they are created from the `core:notifier` module or from the cluster.

As discussed, I'll harmonize the way we create those notifications whatever who instantiates them.